CONTRIBUTING.md

```diff
@@ -51,5 +51,104 @@ This process can be performed automatically by running `make license-generator`.
 
 Sometimes pull requests take a little while to be merged. This is partially because they often have knock-on effects for the rest of the ecosystem, and partially because we want to give core team members time to review and consider changes thoroughly. Please see the organization's [governance document](https://github.com/purescript/governance) for information about when a pull request may be merged.
 
+## Developer Guide
+
+The following instructions are intended to help PureScript users more easily contribute to the compiler, even if this is your first Haskell project.
+
+### Prerequisites
+
+Install `stack`. [Instructions](https://docs.haskellstack.org/en/stable/README/).
+
+See the [PureScript Install Guide](INSTALL.md) for information on the `curses`/`ncurses` dependency.
+
+Update stack's package index before proceeding:
+```
+stack update
+```
+
+### Clone
+
+```
+git clone https://github.com/purescript/purescript.git purescript_compiler
+cd purescript_compiler
+```
+
+### Build
+
+```
+stack build
+```
+
+This will take a while the first time it is run.
+
+### Install
+
+If you'd like to test the build in a local shell, you can run:
+```
+stack exec bash
+```
+
+This prepends the location of the compiler to your `$PATH`, so you can test your changes. This works with other tools that depend on the `purs` binary, such as `spago`. Note that rebuilds continue to use this same path, so you don't need to keep re-running `stack exec` to test new changes (assuming you're using a separate terminal for `stack exec`). Remember that `stack exec` must be launched from the same directory that you ran `stack build` from.
+
+You may also install globally with:
+```
+stack install
```

Comment:
I'm always hesitant to use this command because of the global installation. Shouldn't this be covered in `INSTALL.md` instead?

Reply:
I figure some contributors are editing the compiler to fix an issue that's bothering them, and they may want to continue using using their patched version in the meantime while waiting for the next release (without having to run `stack exec` in each new shell). At least that was my use case when writing this guide.

My interpretation of `INSTALL.md` is that it's for folks who just want to grab the latest official version. It seems more efficient to let these two lines live here, rather than edit the [related section](https://github.com/purescript/purescript/blob/master/INSTALL.md#compiling-from-source) of that guide to also include additional instructions for installing untagged versions.

Reply:
Good point. Maybe add a note here that one should use this 'global install' only temporarily for the purpose you mentioned above and that they can remove it by running `stack path --local-bin` to see which folder stores the binary and delete the binary once done?

Reply:
I think `which purs` beats `stack path --local-bin` for tracking down binaries. Added a note about that utility to this and the previous section.

Reply:
Good idea!